### PR TITLE
perf: use native URL when available

### PR DESCRIPTION
**Pull request by gkatsev**
@gesinger did a quick profile and usage when from ~30% to ~3% of time spent doing work.

This is now fully backwards compatible across all platforms.

### CODE REVIEW

**Comment by gkatsev** [comment]
Assuming, we're fine with converting protocol relative urls to absolute urls (I don't see an issue, frankly), then we can keep 91b966b988402acd934a6a4ee88fe40815e9797a, otherwise, we'll want to do something extra.

**Comment by gkatsev** [comment]
This, the change now works in browsers but it now fails in when running `npm run test:node`

**Review by brandonocasey — APPROVED** [review]
Tested linked into vhs and everything appears to work. I feel like I notice a speedup in the vhs unit tests too 👍🏻

File: src/resolve-url.js [hunk 1]
```diff
@@ -7,12 +7,38 @@ const resolveUrl = (baseUrl, relativeUrl) => {
...
   }
 
+  if (nativeURL) {
+    const newUrl = new URL(relativeUrl, baseUrl);
+
+    // if we're a protocol-less url, remove the protocol
+    // and if we're location-less, remove the location
+    // otherwise, return the url unmodified
+    if (removeLocation) {
+      return newUrl.href.slice(18);
```
Comment (gesinger) [inline src/resolve-url.js:33]: Probably worth adding `const DEFAULT_LOCATION = 'http://example.com'` and DEFAULT_LOCATION.length

**Review by gesinger — APPROVED** [review]

**Pull request merged**